networkit/cpp/graph/test/GraphToolsGTest.cpp

```diff
@@ -1023,4 +1023,36 @@ TEST_P(GraphToolsGTest, testEdgesSortedByWeight) {
         EXPECT_TRUE(G.checkConsistency());
     }
 }
+
+TEST_P(GraphToolsGTest, testTopologySort) {
+    auto G = Graph(5, false, directed());
+
+    /**
+     * /--> 1 --> 3
+     * |    ^
+     * 0    |
+     * |    |
+     * \--> 2 <-- 4
+     */
+
+    // One valid topology is 4,0,2,1,3
+    G.addEdge(0, 1);
+    G.addEdge(0, 2);
+    G.addEdge(2, 1);
+    G.addEdge(1, 3);
+    G.addEdge(4, 2);
+
+    if (directed()) {
+        std::vector<node> res = GraphTools::topologySort(G);
+        std::vector<node> test = {4, 0, 2, 1, 3};
+        EXPECT_TRUE(res == test);
```

Comment:
This test looks rather unstable, as there are multiple possible results, and the actual returned order depends on the implementation, the actual choice from the possible results is undefined behaviour. 

The test could instead iterate through all edges *(u, v)* and assert that u's index in the result is smaller than v's index.